cmd/hatchet-migrate/migrate/migrations/20250407012633_v1_0_11.sql

```diff
@@ -0,0 +1,749 @@
+-- +goose Up
+-- +goose StatementBegin
+CREATE OR REPLACE FUNCTION after_v1_concurrency_slot_insert_function()
+RETURNS trigger AS $$
+BEGIN
+    WITH parent_slot AS (
+        SELECT
+            *
+        FROM
+            new_table cs
+        WHERE
+            cs.parent_strategy_id IS NOT NULL
+    ), parent_to_child_strategy_ids AS (
+        SELECT
+            wc.id AS parent_strategy_id,
+            wc.tenant_id,
+            ps.workflow_id,
+            ps.workflow_version_id,
+            ps.workflow_run_id,
+            MAX(ps.sort_id) AS sort_id,
+            MAX(ps.priority) AS priority,
+            MAX(ps.key) AS key,
+            ARRAY_AGG(DISTINCT wc.child_strategy_ids) AS child_strategy_ids
+        FROM
+            parent_slot ps
+        JOIN v1_workflow_concurrency wc ON wc.workflow_id = ps.workflow_id AND wc.workflow_version_id = ps.workflow_version_id AND wc.id = ps.parent_strategy_id
+        GROUP BY
+            wc.id,
+            wc.tenant_id,
+            ps.workflow_id,
+            ps.workflow_version_id,
+            ps.workflow_run_id
+    )
+    INSERT INTO v1_workflow_concurrency_slot (
+        sort_id,
+        tenant_id,
+        workflow_id,
+        workflow_version_id,
+        workflow_run_id,
+        strategy_id,
+        child_strategy_ids,
+        priority,
+        key
+    )
+    SELECT
+        pcs.sort_id,
+        pcs.tenant_id,
+        pcs.workflow_id,
+        pcs.workflow_version_id,
+        pcs.workflow_run_id,
+        pcs.parent_strategy_id,
+        pcs.child_strategy_ids,
+        pcs.priority,
+        pcs.key
+    FROM
+        parent_to_child_strategy_ids pcs
+    ON CONFLICT (strategy_id, workflow_version_id, workflow_run_id) DO NOTHING;
```

Comment:
what's the reason for not updating here?

Reply:
The only field that we were previously updating was the `completed_child_strategy_ids` which is now deprecated (will add a comment). 

I'm wondering if there's an extremely rare edge case where we end up hitting the `ON CONFLICT` case concurrently with another query that's decided to delete the `v1_workflow_concurrency_slot` -- even though that query takes out a `FOR UPDATE` lock. I think `ON CONFLICT` acquires a `RowExclusiveLock` on the the primary key (https://git.postgresql.org/gitweb/?p=postgresql.git;a=blob;f=src/backend/executor/execIndexing.c;h=1f1181b56044a599398433a0f7feb45206a28abd;hb=HEAD) but I'm not 100% sure. 

In the worst case, however, we end up deleting the concurrency slot early, which is better than the alternative where the concurrency slot is never deleted.